esmvaltool/diag_scripts/ocean/diagnostic_timeseries.py

```diff
@@ -62,11 +63,65 @@ def timeplot(cube, **kwargs):
 
     Needed because iris version 1.13 fails due to the time axis.
     """
-    if iris.__version__ > '2.0':
-        qplt.plot(cube, kwargs)
-    else:
-        times = diagtools.timecoord_to_float(cube.coord('time'))
-        plt.plot(times, cube.data, **kwargs)
+    cubedata = np.ma.array(cube.data)
+    if len(cubedata.compressed()) == 1:
+        plt.axhline(cubedata.compressed(), **kwargs)
+        return
+
+    times = diagtools.cube_time_to_float(cube)
+    plt.plot(times, cubedata, **kwargs)
+
+
+def moving_average(cube, window):
+    """
+    Make a moving average plot.
+
+    the window is a string which isa number and a measuremet of time.
```

Comment:
can you give an example of `window` in the docstring pls?

Reply:
Changed the docstring to:
```
      Calculate a moving average.

      The window is a string which is a number and a measuremet of time.
      For instance, the following are acceptable windows:
          '5 days'
          '12 years'
          '1 month'
          '5 yr'

      Also note the the value used is the total width of the window.
      For instance, if the window provided was '10 years', the the moving
      average returned would be the average of all values within 5 years
      of the central value.
      In the case of edge conditions, at the start an end of the data, they
      only include the average of the data available. Ie the first value
      in the moving average of a '10 year' window will only include the average
      of the five subsequent years.
```